libyul/AsmAnalysis.cpp

```diff
@@ -530,10 +530,11 @@ void AsmAnalyzer::checkAssignment(Identifier const& _variable, YulString _valueT
 	bool found = false;
 	if (Scope::Identifier const* var = m_currentScope->lookup(_variable.name))
 	{
-		// Check that it is a variable
-		if (!holds_alternative<Scope::Variable>(*var))
-			m_errorReporter.typeError(2657_error, _variable.location, "Assignment requires variable.");
-		else if (!m_activeVariables.count(&std::get<Scope::Variable>(*var)))
+		// Check that it is a variable.
+		// This can also hold a function, but that is caught by error 6041.
```

Comment:
So `var` can hold `Scope::Variable` or `Scope::Function`. Making it hold function is not possible, because that is caught by 6041. The test case would be:
```
function f() {}
let a := f
```

6041 says `f` must be called.

Reply:
I think var can still point to a Yul function. Please see #10975